test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_string.phpt

```diff
@@ -0,0 +1,158 @@
+
+    if (AE\isColEncrypted()) {
+		// Create a Store Procedure
+		$spname = 'selectAllColumns';
+		createProc($conn, $spname, "@c_det $dataType OUTPUT, @c_rand $dataType OUTPUT", "SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname");
+	}
+	
+    // insert a row
+    $inputValues = array_slice(${explode("(", $dataType)[0] . "_params"}, 1, 2);
+    $r;
+    $stmt = AE\insertRow($conn, $tbname, array( $colMetaArr[0]->colName => $inputValues[0], $colMetaArr[1]->colName => $inputValues[1] ), $r);
+    if ($r === false) {
+        is_incompatible_types_error($dataType, "default type");
+    }
+	
+    foreach($directions as $direction) {
+        echo "Testing as $direction:\n";
+
+        // test each SQLSRV_SQLTYPE_ constants
+        foreach ($sqlTypes as $sqlType) {
+            if (!AE\isColEncrypted()) {
+                $isCompatible = false;
+                foreach ($compatList[$dataType] as $compatType) {
+                    if (stripos($compatType, $sqlType) !== false) {
+                        $isCompatible = true;
+                    }
+                }
+                // 22018 is the SQLSTATE for any incompatible conversion errors
+                if ($isCompatible && sqlsrv_errors()[0]['SQLSTATE'] == 22018) {
+                    echo "$sqlType should be compatible with $dataType\n";
+                    $success = false;
+                }
+            } else {
+                // skip unsupported datetime types
+                if (!isDateTimeType($sqlType)) {
+                    $sqlTypeConstant = get_sqlType_constant($sqlType);
+
+                    // Call store procedure
+                    $outSql = AE\getCallProcSqlPlaceholders($spname, 2);
+                    $c_detOut = '';
+                    $c_randOut = '';
+                    $stmt = sqlsrv_prepare($conn, $outSql, 
+                        //array(array(&$c_detOut, constant($direction), null, $sqlTypeConstant),
+                        //array(&$c_randOut, constant($direction), null, $sqlTypeConstant)));
+                        array(array(&$c_detOut, SQLSRV_PARAM_INOUT, null, $sqlTypeConstant),
+                        array(&$c_randOut, SQLSRV_PARAM_INOUT, null, $sqlTypeConstant)));
+
+                    if (!$stmt) {
+                        die(print_r(sqlsrv_errors(), true));
+                    }						
+                        
+                    sqlsrv_execute($stmt);
+                    $errors = sqlsrv_errors();
+                    
+                    if (!empty($errors) ) {
+                        if (stripos("SQLSRV_SQLTYPE_" . $dataType, $sqlType) !== false) {
+                            var_dump(sqlsrv_errors());
+                            $success = false;                    
+                        }                
+                    }
+                    else
+                    {
+                        print("c_det: " . $c_detOut . "\n");
+                        print("c_rand: " . $c_randOut . "\n");

```

Comment:
again I don't think printing them is necessary as they are already being compared against the inputValues